test/server/cards/02_SHD/units/RicketyQuadjumper.spec.ts

```diff
@@ -0,0 +1,90 @@
+describe('RicketyQuadjumper', function () {
+    integration(function (contextRef) {
+        describe('Rickety Quadjumper\'s ability', function () {
+            it('should give an experience token to another unit if the revealed card is not a unit', function () {
+                const { context } = contextRef;
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['wampa', 'battlefield-marine'],
+                        spaceArena: ['rickety-quadjumper'],
+                        deck: ['protector'],
+                    },
+                    player2: {
+                        groundArena: ['atst'],
+                    }
+                });
+
+
+                // attack with rickety
+                context.player1.clickCard(context.ricketyQuadjumper);
+                context.player1.clickCard(context.p2Base);
+
+                // player1 should have prompt or pass
+                expect(context.player1).toHavePassAbilityPrompt('Reveal a card');
+                context.player1.clickPrompt('Reveal a card');
+
+                // top card is an upgrade, give exp to another unit
+                expect(context.protector).toBeInZone('deck');
```

Comment:
Need to check that the reveal showed the expected card in the chat log. See usages of `getChatLogs()` in the C-3PO tests for an example